### DISCUSSION TITLE
trame-cookiecutter Components option

### DISCUSSION THREAD

**Discussion by earlsioson** [Kitware/trame#90] [Q&A]
Hello,

@jourdain is the trame-cookiecutter for Component option for v2 working? I can get the App version working both with the image and server docker scripts and with pvpython. But none of those works when I try with selecting the Components option.

**Comment by jourdain** [comment]
Did you build the JS code like described in the README of the generated project?

**Reply by earlsioson** [reply]
Yes I did. The transpiled js was added to the module/serve directory.

**Reply by psavery** [reply]
"App with Components" seems to work for me in a fresh environment. See the image below.

![Screenshot from 2022-07-12 10-04-54](https://user-images.githubusercontent.com/9558430/178509154-9afaf5dd-509a-4866-9829-1478cf801aff.png)

**Reply by earlsioson** [reply]
thank you @psavery, I try again with latest.

**Reply by jourdain** [reply]
Is that still an issue? Or is it working now?

**Reply by earlsioson** [reply]
@jourdain, I tried again this time in Ubuntu and it worked.

I wasn't able to get it to work on my Mac though. I think it's because python on my mac is v3.9.13

**Reply by jourdain** [reply]
Thanks! It might be a pip version which prevent you from getting a vtk9+ ?

**Comment by jourdain** [comment]
Today I ran into a similar situation where the widgets would not build/work because __vue@2.7__ was getting picked.
Forcing the version of __vue-template-compiler__ and __vue__ to be __2.6.14__ solved the issue.

I'm not sure this is related but I wanted to mention it.

(Fixed the version of vue, just had a typo)

**Reply by earlsioson** [reply]
thanks for mentioning @jourdain, I tried again on my Mac  with your suggestion but the highest version I could pin vue and vue-template-compiler is at 2.6.14...

<img width="877" alt="image" src="https://user-images.githubusercontent.com/13466524/180334890-2903d30c-646d-4a53-9ba8-2921244e0b5e.png">

but when I rebuilt with 2.6.14 (I confirmed by checking version in node_modules), it still doesn't work. I get this error...

<img width="1319" alt="image" src="https://user-images.githubusercontent.com/13466524/180335033-db51e6ea-78b3-48c9-a841-4db9cf26aae9.png">

For now we are getting by using an ubuntu container since it works ok there.

**Reply by jourdain** [reply]
The error point to the fact that the JS/CSS files don't get properly served. Are you sure the `npm run build` completed without issue using 2.6.14? Also in your venv did your `pip install .` or `pip install -e .`? If you didn't put the `-e` you must execute it once the `npm run build` succeeded. 

HTH
